pkg/services/ngalert/ngalert.go

```diff
@@ -218,7 +218,7 @@ func (ng *AlertNG) init() error {
 						ExternalURL:       ng.Cfg.AppURL,
 						StaticHeaders:     ng.Cfg.Smtp.StaticHeaders,
 					}
-					remoteAM, err := createRemoteAlertmanager(cfg, ng.KVStore, ng.SecretsService.Decrypt, autogenFn, m, ng.tracer)
+					remoteAM, err := createRemoteAlertmanager(ctx, cfg, ng.KVStore, ng.SecretsService.Decrypt, autogenFn, m, ng.tracer)
```

Comment:
This function now takes a context, which we need to call `autogenFn()` and get a default config with auto-gen routes.